src/variety/cbblock.js

```diff
@@ -38,6 +74,17 @@ Border:{
 	isGround : function(){ return (this.ques>0);}
 },
 
+"Border@dbchoco":{
+	isGround : function(){
+		if(this.sidecell[0].isnull || this.sidecell[1].isnull) {return false;}
+		return this.qans === 0 && this.sidecell[0].ques === this.sidecell[1].ques;
+	},
+
+	posthook: {
+		qans: function(num) { this.board.tilegraph.rebuild(); }
```

Comment:
while it's probably not actually too noticeably expensive, it still feels wrong to rebuild the full graph here... (and yes, I know from experience trying to build scrin with a two-level graph that that's not necessarily something that will be avoidable)

anyway fine to leave it as is at this point, but I want to think about whether there's a way to improve this still

Reply:
After some consideration, I've found that the AreaTileGraph and the AreaBlockGraph aren't actually related, but the extra information set in AreaBlockGraph nodes depend on an up to date AreaTileGraph.

I've removed all calls to `rebuild`. When the AreaTileGraph changes, the `dotcnt` is now updated only in the affected blocks. This also fixes a problem in the master version of the cbblock editor, where drawing dotted lines (and not solution borders) would always return the error "A block has one area framed by dotted line."

Reply:
looks good, I pulled the change to #45 and added a test